From 2f5d4b7633865a1fc3f4117d1ec8c0b31e1fabbe Mon Sep 17 00:00:00 2001
From: atomlong <atom.long@hotmail.com>
Date: Sun, 24 Oct 2021 12:11:58 +0800
Subject: [PATCH] fix buffer size If the second argument to realpath is not
 NULL, it must be a buffer at least PATH_MAX bytes long. [1]
 https://man7.org/linux/man-pages/man3/realpath.3.html

---
 src/main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index d6f2a81..b461ccd 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -35,11 +35,11 @@ void SetConsoleTitle(const std::string &title)
 
 void setcd(std::string &file)
 {
-    char szTemp[1024] = {}, filename[256] = {};
+    char szTemp[PATH_MAX] = {}, filename[NAME_MAX] = {};
     std::string path;
 #ifdef _WIN32
     char *pname = NULL;
-    DWORD retVal = GetFullPathName(file.data(), 1023, szTemp, &pname);
+    DWORD retVal = GetFullPathName(file.data(), PATH_MAX-1, szTemp, &pname);
     if(!retVal)
         return;
     strcpy(filename, pname);
-- 
2.33.1.windows.1

